### PR TITLE
chore: Treat internal event differently, primarely do not display info logs for those events

### PR DESCRIPTION
**Pull request by adrien2p**
FIXES TRI-179

**What**
Add support for `internal` options as part of the emitted events. The internal can be treated differently from the classic events. As a first step, we are treating internal event in a way where we do not display info log about them such as the number of subscribers for a particular internal event. Though, we do still log warning in case someone still find out about an event and get an error from the handler

### CODE REVIEW

**Review by olivermrbl — COMMENTED** [review]
Question: Since `@EmitEvents` decorator will likely only be used module services, would it be reasonable to default to `internal: true`?

**Comment by adrien2p** [comment]
> Question: Since `@EmitEvents` decorator will likely only be used module services, would it be reasonable to default to `internal: true`? 

Well, originally i went with approximately what you are suggesting and then thought that a user could use the decorator and not mark it as internal and it would make the dx weird to have to set it to false every time you use the decorator. But from my understanding this is temporary and also maybe some logs can be removed as it is currently not really aligned across the event bus modules. My takes would be to specify it on our side only until we reach a point of certainty and then see what we do and in the meantime we dont change the users dx

**Comment by olivermrbl** [comment]
My thinking was to not allow users to use module events for the time being. I feel like it's better to hide the concept entirely until we know how to deal with them. Having the `internal: true` as a default would mean they would never stumble into these events.

**Review by olivermrbl — APPROVED** [review]
LGTM

**Comment by adrien2p** [comment]
@olivermrbl do you want to have a last review before merging ?

File: packages/modules/event-bus-local/src/services/event-bus-local.ts
Comment (olivermrbl) [inline packages/modules/event-bus-local/src/services/event-bus-local.ts]: what's the case here?

Comment (adrien2p) [reply packages/modules/event-bus-local/src/services/event-bus-local.ts]: Options can be passed to the emit or as part of the message, so we check both

File: packages/modules/event-bus-local/src/services/event-bus-local.ts
Comment (adrien2p) [inline packages/modules/event-bus-local/src/services/event-bus-local.ts]: I need to fix this

**Comment by olivermrbl** [comment]
Let's merge this when we are ready to publish new release notes. Would like to highlight these changes, as it could otherwise be a bit disruptive for end-users.

**Pull request merged**
